tests/system/common.py

```diff
@@ -780,6 +781,69 @@ def task_by_name(tasks, name):
 
     assert False, "Did not find task with name %s in this list of tasks: %s" % (name, tasks,)
 
+def assert_app_in_all_domains(app, regions=None, zones=None):
+    """Asserts that all tasks of the given app are within all of the given region and/or zone
+    :param app_id: The app instance, as obtained by `marathon_client.get_app(...)`
+    :param region: The regions to test (can be None, String, List or Tuple)
+    :param zone: The zones to test (can be None, String, List or Tuple)
+    """
+
+    if type(regions) is str:
+        regions = [regions]
+    if type(zones) is str:
+        zones = [zones]
+
+    # Sanitize agent domains and user input
+
+    slave_domains = get_cluster_slave_domains()
+
+    assert len(slave_domains) > 0, "Did not find any agents in the DC/OS cluster" % (app['id'],)
+
+    if regions is not None:
+        slave_regions = set([x.region for x in slave_domains.values()])
+        unknown_regions = set(regions) - slave_regions
+        assert len(unknown_regions) == 0, "Region(s) %s was not found in the cluster (expecting one of %s)" \
+                              % (', '.join(unknown_regions), ', '.join(slave_regions))
+    if zones is not None:
+        slave_zones = set(map(lambda x: x.zone, slave_domains.values()))
+        unknown_zones = set(zones) - slave_zones
+        assert len(unknown_zones) == 0, "Zone(s) %s was not found in the cluster (expecting one of %s)" \
+                                             % (', '.join(unknown_zones), ', '.join(slave_zones))
+
+    # Check if regions and/or zones overlap completely
+
+    (used_regions, used_zones) = get_used_regions_and_zones(get_app_domains(app))
+
+    if regions is not None:
+        assert used_regions == set(regions), \
+            "Application %s is not running on all the exepected regions ({%s} instead of {%s})" % (
+                app['id'], ', '.join(used_regions), ', '.join(regions)
+            )
+
+    if zones is not None:
+        assert used_zones == set(zones), \
+            "Application %s is not running on all the exepected zones ({%s} instead of {%s})" % (
+                app['id'], ', '.join(used_zones), ', '.join(zones)
+            )
+
+
+def assert_app_in_some_domains(app, region=None, zone=None):
+    """Asserts that any of the tasks of the given app are within all of the given region and/or zone
```

Comment:
Not sure what this means. Can a task be in all regions at the same time?

Reply:
This function was not used. I removed it.